ls1intum/Artemis
# `General`: Fix LTI 1.3 launch redirect bug

### Checklist
#### General
#### Client

### Motivation and Context
The LTI1.3 launch has a bug where the user is not redirected to the exercise page because the URL to redirect the user to contains queryParams and therefore no routes are found matching the URL. The user stays stuck in the "Launching..." page. This was broken by https://github.com/ls1intum/Artemis/pull/5948

### Description
NavigateByUrl does not work with absolute URLs including queryParams, window.location.replace has to be used instead.

### Steps for Testing
Prerequisites:
- 1 Instructor in Artemis
- 1 Instructor in an external LMS (https://moodle.ase.in.tum.de/my/courses.php)

1. Setup an online course in Artemis as the instructor
2. Create an exercise for the course
3. Setup an LTI exercise in the external LMS
4. Verify that launching the exercise works, and you get redirected to the exercise page (if a new user, otherwise you get redirected to the login page)

Example:

https://user-images.githubusercontent.com/11006274/215709626-160adae1-dd53-4d56-983f-2500be9832e5.mp4

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| lti13-exercise-launch.component.ts | 85% | 100% |

### Screenshots
Before:
![image](https://user-images.githubusercontent.com/11006274/213942736-b77e54cf-b230-415c-bfb0-6aac8ea716f0.png)
